index.js

```diff
@@ -1,27 +1,112 @@
 const { text } = require("./shiroko.json").parse;
 
 function getStudent() {
-	const {
-		groups: { studentName },
-	} = /class=\"character-name character-header\"\s?title=\".*\">(?<studentName>\w*\s?_?\(?\w*\)?)/gim.exec(
-		text
-	);
+
+	const studentName = (() => {
+		const { groups: { en, jp }, } =
+			(/<th>Full Name<\/th><td>(?<en>[\w\p{L}\s]*)(?:<br \/>|\s*)\((?<jp>[\w\p{L}\s]*)\)<\/td>/gisu.exec(
+				text
+			));
+		return { en, jp };
+	})();
 
 	const {
 		groups: { studentRarity },
 	} = /class=\"character-rarity\" data-value=\"(?<studentRarity>\d)/gim.exec(text);
 
 	const {
 		groups: { studentProfileImg },
-	} =
-		/data-title=\"Profile\s?Image[\"><a-z\s=\-:\/\.]*src=\"\/\/(?<studentProfileImg>[a-zA-Z\.\/0-9-_]*)/gim.exec(
+	} = /data-title=\"Profile\s?Image[\"><a-z\s=\-:\/\.]*src=\"\/\/(?<studentProfileImg>[a-zA-Z\.\/0-9-_]*)/gim.exec(
+		text
+	);
+	
+	const {
+		groups: { studentSchool, studentGroup, studentRole },
+	} = /(?:<th class=\"character-school character-header\">.*?<td title=\")(?<studentSchool>[^,]*?)(?:,\s*)(?<studentGroup>[^\"]*)(?:\">)(?:.*?<td>.*?<\/p>\s*)(?<studentRole>.*?)(?:<\/td>)/gis.exec(
+		text
+	);
+	
+	const {
+		groups: { studentDamageType },
+	} = /(?:<tr><th>Damage Type<\/th>\s*<td.*?>)(?<studentDamageType>[^<]*)(?:<\/td>)/gis.exec(
+		text
+	);
+
+	const {
+		groups: { studentArmorType },
+	} = /(?:<tr><th>Armor Type<\/th>\s*<td.*?>)(?<studentArmorType>[^<]*)(?:<\/td>)/gis.exec(
+		text
+	);
+
+	const {
+		groups: { studentCombatClass },
+	} = /(?:<tr><th>Combat Class<\/th>\s*<td.*?>)(?<studentCombatClass>[^<]*)(?:<\/td>)/gis.exec(
+		text
+	);
+
+	const studentAffinities = (() => {
+		let [urban, field, indoors] = text.matchAll(/(\w+)(?: grade affinity)/gis).map(a => a[1]);
```

Comment:
As matchAll() returns an Iterable, map() isn't a function that works on it, so it throws an error when used. Since map() works on Arrays, spreading the Iterable inside an Array works
```suggestion
		let [urban, field, indoors] = [... text.matchAll(/(\w+)(?: grade affinity)/gis)].map(a => a[1]);
```